### PR TITLE
Add `createMetric` and `createSegment` e2e commands

### PR DESCRIPTION
**Pull request by oisincoveney**
### Description

Added `createMetric` and `createSegment` commands that I'll be using in #32392. 

### How to verify

All tests should still pass

### CODE REVIEW

File: e2e/support/commands/api/metric.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+Cypress.Commands.add(
```
Comment (WiNloSt) [inline e2e/support/commands/api/metric.js:1]: I'm not sure `metric` and `segment` should have their own entities. From this admin page, they're under **Table metadata**
image


[In the documentation](https://www.metabase.com/docs/latest/data-modeling/segments-and-metrics), they're under **Data modeling** which corresponds with the admin tab **Data model** before we changed it to **Table metadata**

It feels to me these 2 entities are closely related and should stay close together possibly within the same file. I know their API endpoints make them look like they're completely different entities. But they always exist next to next as far as I can tell.

What do you think? 

The reason I made this comment is that I have expressed my concern about [test util here.](https://github.com/metabase/metabase/pull/32433#discussion_r1272327860) tl;dr I believe we should really think hard about introducing new E2E test utils.

Comment (oisincoveney) [reply e2e/support/commands/api/metric.js:1]: I see what you mean, I found it a bit weird that they both had the same params and things as well. 

I'm approaching this from the search side, where those entities are treated differently when you search for them (i.e. segments and metrics are separate types). In the query builder, they're seen in different places and serve different purposes. While they're in the same area in the settings and are somewhat related, I don't know if we should couple them as much as that

Comment (oisincoveney) [reply e2e/support/commands/api/metric.js:1]: From my perspective, I think we should use the commands as a way to abstract the API calls away from people writing the E2E tests, along with reducing duplicated code etc etc. What do you think? 

I'm happy to combine the files into a `segment-metric.js` but I just don't know if that makes sense

Comment (kamilmielnik) [reply e2e/support/commands/api/metric.js:1]: @oisincoveney Please see these discussions:
- https://github.com/metabase/metabase/pull/32139#discussion_r1253171620
- https://metaboat.slack.com/archives/C057T1QTB3L/p1688566347779349

I believe these should be simple functions, which is what I ended up doing here: #32139.

Comment (oisincoveney) [reply e2e/support/commands/api/metric.js:1]: Yeah, those discussions make sense, but there's a difference between the custom Chai assertions in those discussions and the code in this PR, which is just the same thing for `Metric` and `Segment` entities that we already have for `Question` (`cy.createQuestion`), `Collection` (`cy.createCollection`), `Dashboard` (`cy.createDashboard`), `Alert` (`cy.createAlert`), and `Pulse` (`cy.createPulse`). 

Creating a function wouldn't be an issue, but for the sake of consistency and happiness for anyone that wants to create an entity in an E2E test, I think we really should stick to one pattern

Comment (oisincoveney) [reply e2e/support/commands/api/metric.js:1]: @ranquild what do you think? Should this be the beginning of using regular functions for functionality like this, or should we keep using the 'cy.create<whatever>' pattern?

Comment (ranquild) [reply e2e/support/commands/api/metric.js:1]: @oisincoveney I think we have plain actions for actions actions already

Comment (oisincoveney) [reply e2e/support/commands/api/metric.js:1]: Gotcha, I'll switch it then.

**Review by ranquild — APPROVED** [review]

**Pull request merged**